tt_metal/detail/tt_metal.hpp

```diff
@@ -284,7 +284,9 @@ namespace tt::tt_metal{
             static std::mutex cq_creation_mutex;
             {
                 std::lock_guard<std::mutex> lock(cq_creation_mutex);
-                command_queues[device->id()] = std::make_unique<CommandQueue>(device, 0);
+                if (not command_queues[device->id()] or (command_queues[device->id()] and command_queues[device->id()]->device != device)) {
```

Comment:
I know this was what was previously on main but during your multi-cq work I think you ran into an issue here a device was opened then closed then opened again and in that sequence a new sys mem manager was created but the command queue did not have a valid sysmem manager. 

If you need this pr merged asap then perhaps add a comment and open an issue up for that?